thirdeye/thirdeye-frontend/app/utils/manage-alert-utils.js

```diff
@@ -175,46 +176,67 @@ export function evalObj() {
   };
 }
 
+/**
+ * Builds the request parameters for the metric data API call.
+ * @method buildMetricDataUrl
+ * @param {Object} graphConfig - the metric settings
+ * @returns {String} metric data call params/url
+ */
+export function buildMetricDataUrl(graphConfig) {
+  const { id, maxTime, filters, dimension, granularity } = graphConfig;
+  const selectedDimension = dimension ? dimension : 'All';
+  const startTimeBucket = granularity && granularity.toLowerCase().includes('minute') ? 'week' : 'months';
```

Comment:
This seems really dense for me. Any chance we can break this up? Is `startTimeBucket` supposed to be a boolean?

Reply:
Adding a note for inline documentation on each step.